developer_manual/app/js.rst

```diff
@@ -46,3 +46,37 @@ Full URLs can be generated by using:
 .. code-block:: js
 
     var authorUrl = OC.generateUrl('/apps/myapp/authors/1');
+
+
+Extending core parts
+====================
+
+It is possible to extend components of the core web UI. The following examples
+should show how this is possible.
+
+Extending the "new" menu in the files app
+-----------------------------------------
+
+.. versionadded:: 9.0
+
+.. code-block:: js
+
+    var myNewFileEntry = {
+        attach: function (menu) {
+            menu.addMenuEntry({
+                id: 'abc',
+                displayName: 'Menu display name',
+                templateName: 'templateName.ext',
+                iconClass: 'icon-filetype-text',
+                fileType: 'file',
+                actionHandler: function () {
+                    console.log('do something here');
+                }
+            });
+        }
+    };
+    OC.Plugins.register('OCA.Files.NewFileMenu', myNewFileEntry);
```

Comment:
may I suggest renaming this to `myFileMenuPlugin` instead ?
Because at this point you can add several entries.

Optionally you could also hack/override/extend the existing function classes to make them do additional stuff.